### DISCUSSION TITLE
How to add an icon to the TOC to indicate that it is an external link?

### DISCUSSION THREAD

**Discussion by eyllanesc** [squidfunk/mkdocs-material#3595] [Q&A]
In many sphinx themes an icon is placed to indicate that it is an external link to the documentation: 

![Screenshot_20220218_110808](https://user-images.githubusercontent.com/4896200/154719739-1f4163e7-cb50-4bd2-babb-7f24ee118f3e.png)

How can I do something similar in mkdocs?

**Comment by Stanzilla** [comment]
Hey! I use something like this in a custom javascript file:

```js
function addListenerMulti(element, eventNames, listener) {
  const events = eventNames.split(" ")
  for (let i = 0, iLen = events.length; i < iLen; i++) {
    element.addEventListener(events[i], listener, false)
  }
}

addListenerMulti(document, "DOMContentLoaded DOMContentSwitch", (event) => {
  // Add an icon to all external links
  const links = document.querySelectorAll(".md-content a")
  for (let i = 0, length = links.length; i < length; i++) {
    if (
      links[i].hostname !== window.location.hostname &&
      links[i].hostname !== "" &&
      links[i].classList.contains("md-icon") !== true &&
      links[i].classList.contains("download-button") !== true
    ) {
      links[i].target = "_blank"
      links[i].classList.add("external")
      links[i].rel = "noopener"
    }
  }
})
```

**Reply by ssbarnea** [reply]
Using a custom JS file to implement such a feature looks like overkill to me, especially as I do remember that I implemented something like this in the past using pure CSS.

**Comment by northword** [comment]
@Stanzilla 's work is excellent, many thanks! [Here is an example](https://github.com/zotero-cn/csl-doc-chinese/blob/d07efcab21ec93c76571dafece99e54cd053a642/docs/assets/js/custom.js) i based on him, it looks pretty good! 

It is worth noting that when using this `js`, you have to disable the `navigation.instant` to ensure that the icon can be added in the newly entered page without refreshing. I am not sure if there is any callback function or solusion to solve this.

<details>
<summary>Results display</summary>

![Results display](https://user-images.githubusercontent.com/44738481/182860214-da670fda-c145-4803-b7cd-82e80e1d4a52.png)

</details>

**Comment by z-aki** [comment]
created a plugin for this https://github.com/z-aki/mkdocs-plugin-open-external-links-in-new-tab
